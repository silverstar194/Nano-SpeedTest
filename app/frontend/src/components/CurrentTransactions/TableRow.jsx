import React, {Fragment} from 'react';
import PropTypes from 'prop-types';

const TableRow = ({
    origin,
    destination,
    startSendTimestamp,
    endSendTimestamp,
    error,
    completed,
    amount,
    transactionHashReceiving,
    transactionHashSending
}) => {
    const classStyle = error ? 'table-danger' : '';
    return (
        <tr className={classStyle}>
            <td>{origin.nodeLocation}</td>
            <td>{destination.nodeLocation}</td>
            <td>{amount} nano</td>
            { completed ?
                <Fragment>
                { error ?
                    <Fragment>
                        <td>NO DATA</td>
                        <td>ERROR</td>
                        <td>NO DATA</td>
                        <td>NO DATA</td>
                    </Fragment> : <Fragment>
                        <td>{(endSendTimestamp - startSendTimestamp)/1000} Seconds</td>
                        <td>Completed</td>
                        <td className='block-col text-ellipsis'>
                            <a
                                href={`https://www.nanode.co/block/${transactionHashSending}`}
                                target='_blank'
                                rel='noopener noreferrer'>
                                {transactionHashSending}
                            </a>
                        </td>
                        <td className='block-col text-ellipsis'>
                            <a
                                href={`https://www.nanode.co/block/${transactionHashReceiving}`}
                                target='_blank'
                                rel='noopener noreferrer'>
                                {transactionHashReceiving}
                            </a>
                        </td>
                    </Fragment>
                }
                </Fragment>
                : <Fragment>
                    <td>Pending...</td>
                    <td>Pending...</td>
                    <td>Pending...</td>
                    <td>Pending...</td>
                </Fragment>
            }
        </tr>
    );
};

TableRow.propTypes = {
    origin: PropTypes.object.isRequired,
    destination: PropTypes.object.isRequired,
    time: PropTypes.number,
    amount: PropTypes.string,
    completed: PropTypes.bool,
    endSendTimestamp: PropTypes.number,
    startSendTimestamp: PropTypes.number,
    transactionHashReceiving: PropTypes.string.isRequired,
    transactionHashSending: PropTypes.string.isRequired
};

export default TableRow;