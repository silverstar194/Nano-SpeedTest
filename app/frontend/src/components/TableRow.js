import React from 'react';

const TableRow = ({index, hash, origin, destination, time, amount, completed}) => {
    return (
        <tr>
            <th scope="row">{index}</th>
            <td>{hash}</td>
            <td>{origin}</td>
            <td>{destination}</td>
            <td>{time} Seconds</td>
            <td>{amount}</td>
            <td>{completed}</td>
        </tr>
    );
};

export default TableRow;