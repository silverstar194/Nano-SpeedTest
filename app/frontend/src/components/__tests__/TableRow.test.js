import React from 'react';
import renderer from 'react-test-renderer';
import TableRow from '../TableRow';
import { shallow } from 'enzyme';

const props = {
    index: 0,
    hash: 'abc123',
    origin: 'SF',
    destination: 'LA',
};

it('shallow renders without crashing', () => {
    shallow(<TableRow  {...props}/>);
});

it(`renders snapshot correctly`, () => {
    const tree = renderer.create(
        <TableRow {...props}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
