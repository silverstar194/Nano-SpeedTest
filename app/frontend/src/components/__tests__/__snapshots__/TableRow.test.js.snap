// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<tr>
  <th
    scope="row"
  >
    0
  </th>
  <td>
    abc123
  </td>
  <td>
    SF
  </td>
  <td>
    LA
  </td>
  <td>
     Seconds
  </td>
  <td />
  <td />
</tr>
`;
